### PR TITLE
Porting lifecycle grace period test cases

### PR DESCRIPTION
**Pull request by edcdavid**
In this test case, we look at the yaml applied configuration (found in annotation ) for the  terminationGracePeriodSeconds parameter. If it is not defined then fail the test. The test is done for deployments, statefulsets and unmanaged pods.

### CODE REVIEW

**Review by sebrandon1 — COMMENTED** [review]
Looks good, just a comment about negative test cases.  Thanks!

File: cnf-certification-test/lifecycle/graceperiod/graceperiod_test.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+			wantTerminationGracePeriodSeconds: 30,
+			wantErr:                           false,
+		},
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply cnf-certification-test/lifecycle/graceperiod/graceperiod_test.go:36]: Fixed, added negative cases

File: cnf-certification-test/accesscontrol/rbac/clusterrolebinding_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// // (at your option) any later version.
+// //
+// // This program is distributed in the hope that it will be useful,
+// // but WITHOUT ANY WARRANTY; without even the implied warranty of
+// // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// // GNU General Public License for more details.
+// //
+// // You should have received a copy of the GNU General Public License along
+// // with this program; if not, write to the Free Software Foundation, Inc.,
+// // 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
```
Comment (greyerof) [inline cnf-certification-test/accesscontrol/rbac/clusterrolebinding_test.go:15]: These lines appear as commented twice. Could you leave it with just one, please?

**Review by greyerof — APPROVED** [review]

File: cnf-certification-test/lifecycle/suite.go [hunk 3]
```diff
@@ -178,6 +184,36 @@ func testPodNodeSelectorAndAffinityBestPractices(env *provider.TestEnvironment)
...
 }
+
+func testGracePeriod(env *provider.TestEnvironment) {
+	badDeployments, deploymentLogs := graceperiod.TestTerminationGracePeriodOnDeployments(env)
+	badStatefulsets, statefulsetLogs := graceperiod.TestTerminationGracePeriodOnStatefulsets(env)
+	badPods, podLogs := graceperiod.TestTerminationGracePeriodOnPods(env)
+
+	numDeps := len(badDeployments)
+	if numDeps > 0 {
+		logrus.Tracef("Deployments found without terminationGracePeriodSeconds param set: %+v", badDeployments)
```
[the remark this answers is not in the document]
Comment (edcdavid) [reply cnf-certification-test/lifecycle/suite.go]: fixed. Made the logs debug again.

File: cnf-certification-test/accesscontrol/rbac/clusterrolebinding_test.go (hunk 2 above)
Comment (edcdavid) [reply cnf-certification-test/accesscontrol/rbac/clusterrolebinding_test.go]: fixed

**Review by sebrandon1 — APPROVED** [review]

**Pull request merged**